sdk/azidentity/aad_identity_client.go

```diff
@@ -61,7 +63,37 @@ func newAADIdentityClient(options *TokenCredentialOptions) *aadIdentityClient {
 // clientID: The client (application) ID of the service principal
 // clientSecret: A client secret that was generated for the App Registration used to authenticate the client
 // scopes: The scopes required for the token
-func (c *aadIdentityClient) authenticate(ctx context.Context, tenantID string, clientID string, clientSecret string, scopes []string) (*azcore.AccessToken, error) {
+func (c *aadIdentityClient) refreshAccessToken(ctx context.Context, tenantID string, clientID string, clientSecret string, refreshToken string, scopes []string) (*tokenResponse, error) {
+	msg, err := c.createRefreshTokenRequest(tenantID, clientID, clientSecret, refreshToken, scopes)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.pipeline.Do(ctx, msg)
+	if err != nil {
+		return nil, err
+	}
+
+	// This should never happen under normal conditions
+	if resp == nil {
```

Comment:
I think we should remove this block.  If `Do()` ever returns a nil error and a nil response it's a bug and we should fix it.

Reply:
Are you thinking we should remove this everywhere? (i.e. I also check for this condition for client secret credentials, certificate credentials, etc) I'm not opposed to removing this from all of them.

Reply:
I removed it in all instances, since this should actually never come to pass.